src/event_statistics.cpp

```diff
@@ -690,6 +699,144 @@ struct event_statistic_total : event_statistic::impl {
+            debugmsg( "event_statistic %s refers to field %s in event source %s, but that source "
+                      "has no such field", name, field, source.debug_description() );
+        }
+    }
+
+    cata_variant_type type() const override {
+        return cata_variant_type::int_;
+    }
+
+    monotonically monotonicity() const override {
+        if( is_increasing( source.monotonicity() ) ) {
+            return source.monotonicity();
+        } else {
+            return monotonically::unknown;
+        }
+    }
+
+    std::unique_ptr<impl> clone() const override {
+        return std::make_unique<event_statistic_maximum>( *this );
+    }
+};
+
+struct event_statistic_minimum : event_statistic::impl {
+    event_statistic_minimum( const string_id<event_statistic> &i, const event_source &s,
+                             const std::string &f ) :
+        id( i ), source( s ), field( f )
+    {}
+
+    string_id<event_statistic> id;
+    event_source source;
+    std::string field;
+
+    cata_variant value( stats_tracker &stats ) const override {
+        int minimum = source.get( stats ).minimum( field );
+        return cata_variant::make<cata_variant_type::int_>( minimum );
+    }
+
+    struct state : stats_tracker_state, event_multiset_watcher {
+        state( const event_statistic_minimum *s, stats_tracker &stats ) :
+            stat( s ),
+            value( s->value( stats ).get<int>() ) {
+            stat->source.add_watcher( stats, this );
+        }
+
+        void event_added( const cata::event &e, stats_tracker &stats ) override {
+            const int new_value = std::min( e.get<int>( stat->field ), value );
+            if( new_value != value ) {
+                value = new_value;
+                stats.stat_value_changed( stat->id, cata_variant( value ) );
+            }
+        }
+
+        void events_reset( const event_multiset &new_set, stats_tracker &stats ) override {
+            value = new_set.minimum( stat->field );
+            stats.stat_value_changed( stat->id, cata_variant( value ) );
+        }
+
+        const event_statistic_minimum *stat;
+        int value;
+    };
+
+    std::unique_ptr<stats_tracker_state> watch( stats_tracker &stats ) const override {
+        return std::make_unique<state>( this, stats );
+    }
+
+    void check( const std::string &name ) const override {
+        cata::event::fields_type event_fields = source.fields();
+        auto it = event_fields.find( field );
+        if( it == event_fields.end() ) {
+            debugmsg( "event_statistic %s refers to field %s in event source %s, but that source "
+                      "has no such field", name, field, source.debug_description() );
+        }
+    }
+
+    cata_variant_type type() const override {
+        return cata_variant_type::int_;
+    }
+
+    monotonically monotonicity() const override {
+        if( is_decreasing( source.monotonicity() ) ) {
+            return source.monotonicity();
```

Comment:
This one's wrong.  When the source is increasing (meaning more events added) the `min` will be decreasing.  Currently event sources can't really be decreasing, but I guess if one was then the `min` would be increasing.

This could be written succinctly if we overrode unary `operator-` for `monotonically`.  I can't decide if that's a good idea or an abuse of operator overloading.